include/AST/Types/FunctionTy.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef GLU_AST_TYPES_FUNCTIONTY_HPP
+#define GLU_AST_TYPES_FUNCTIONTY_HPP
+
+#include "TypeBase.hpp"
+#include <vector>
+
+namespace glu::types {
+
+/// @brief FunctionTy is a class that represents a function type in the AST.
+class FunctionTy : public TypeBase {
+private:
+    std::vector<TypeBase *> const _parameters;
+    TypeBase *_returnType;
+
+public:
+    /// @brief Constructor for the FunctionTy class.
+    /// @param parameters A vector of TypeBase pointers representing the
+    /// parameters of the function.
+    /// @param returnType A TypeBase pointer representing the return type of the
+    /// function.
+    FunctionTy(std::vector<TypeBase *> parameters, TypeBase *returnType)
+        : TypeBase(TypeKind::FunctionTyKind)
+        , _parameters(std::move(parameters))
+        , _returnType(returnType)
```

Comment:
I think this should be a kind because we don't know about the return value when the class is being instantiate, only the return type